rust/lance/src/index/append.rs

```diff
@@ -33,7 +33,7 @@ pub async fn merge_indices<'a>(
     dataset: Arc<Dataset>,
     old_indices: &[&'a IndexMetadata],
     options: &OptimizeOptions,
-) -> Result<Option<(Uuid, Vec<&'a IndexMetadata>, RoaringBitmap)>> {
+) -> Result<Option<(Uuid, Vec<&'a IndexMetadata>, RoaringBitmap, i32)>> {
```

Comment:
Minor nit: can we return a simple data struct instead of a 4-tuple?

```
pub struct IndexMergeResult<'a> {
  pub new_id: Uuid,
  pub removed_indices: Vec<&'a IndexMetadata>,
  pub new_frag_ids: RoaringBitmap,
  pub new_index_version: i32,
}
```